src/orwell/AgentProxy.cpp

```diff
@@ -295,7 +295,10 @@ bool AgentProxy::step(
 		std::string aObject;
 		aStream >> aObject;
 		std::string aName;
-		ReadName(aStream, aName);
+		if (("robot" == aObject) or ("team" == aObject))
```

Comment:
I would put this check into a dedicated method to better express its intent.
Eg.: isRobotOrTeam(aObject)